js-packages/@prestojs/viewmodel/src/useViewModelCache.ts

```diff
@@ -1,12 +1,35 @@
-import { useLayoutEffect, useReducer, useRef } from 'react';
+import { useDebugValue, useLayoutEffect, useReducer, useRef } from 'react';
 
 import ViewModelCache from './ViewModelCache';
 import { ViewModelConstructor } from './ViewModelFactory';
 
-type Selector<T extends ViewModelConstructor<any, any>, ResultType> = (
-    cache: ViewModelCache<T>
+type Selector<T extends ViewModelConstructor<any, any>, ResultType, U extends any[]> = (
+    cache: ViewModelCache<T>,
+    ...args: U
 ) => ResultType;
 
+function shallowEqual(a?: any[], b?: any[]): boolean {
+    if (a === b) {
+        return true;
+    }
+    if (!a || !b) {
+        return false;
+    }
```

Comment:
If `a == []` and `b == []` this will return false

Reply:
heh